src/index.js

```diff
@@ -68,66 +66,70 @@ const mapToPropertyDescriptor = (
           return null;
         };
       },
-      set: function(value) {
-        eventHandler = (typeof value === 'function') ? value : null;
-        onAfterSet(this);
+      set(value) {
+        this[name] = (typeof value === 'function') ? value : null;
```

Comment:
You can't properly serialize a function into a string and assign it to an attribute, therefore we have to store it somewhere